src/router_core/router_core.c

```diff
@@ -173,6 +173,9 @@ void qdr_core_free(qdr_core_t *core)
         free(link->insert_prefix);
         free(link->strip_prefix);
         link->name = 0;
+        for (int i = 0; i < QDR_LINK_LIST_CLASSES; ++i) {
+            free_qdr_link_ref_t(link->ref[i]);
```

Comment:
instead of calling free_qdr_link_ref_t is it worth calling 
bool qdr_del_link_ref(qdr_link_ref_list_t *ref_list, qdr_link_t *link, int cls)

Reply:
Good point - it would be better to clean up the ref lists propery.
I'll update this PR shortly...